.github/workflows/ci.yml

```diff
@@ -363,7 +363,7 @@ jobs:
     timeout-minutes: 60
     name: "${{matrix.test-type}}:MySQL${{matrix.mysql-version}}, Py${{matrix.python-version}}"
     runs-on: ubuntu-latest
-    needs: [build-info, ci-images]
+    needs: [build-info, ci-images, tests-postgres]
```

Comment:
So these tests would run sequentially now?

Reply:
Correct. They will run in "per-db" "batches" .  It's not as bad as it seems @dimberman - they will slow down an individual PR if the queue is empty (I think we have up to 100-150 slots in our Queue, not sure exactly). But when there are like 10 PRs (as we currently have) competing for the queue and for example one of them fails in the "static check" phase, there is much higher chance that a new PR or one of those waiting will run "cancel" job much faster and free the queue, which will make the "good" PRs complete faster.

Reply:
Aa discussed with Ash - without solving canceling on "scheduler" level by GitHub, we can at most move between "faster builds" and "snappier cancels" . GitHubs works on it:

```
Hi Jarek,
Thank you for sending this in. We have an open feature request for this and have added your feedback.
I cannot say if or when this will be implemented. Keep an eye on the release notes or the GitHub Blog for information on when/if this makes it into the product.
Cheers,
Olliec
```

And we also might want to have an external job with a token that has only "workflow" scope, canceling the builds.<
I actually thought about this and I might want to do just that - I can modify my "cancel" action slightly to be able to cancel builds from another project (and use a personal token for that with only "workflow" scope and nothing else). That should be rather safe. 

I can run scheduled Github Action workflow from a personal project (so that it does not compete with Apache Queue). Should be rather safe. I only have to take care about rate limiting. Personal projects have 1000 requests/hr limit (much lower than 5000 we have) and I already exceeded it several times when running the cancel action too frequently.

Reply:
If this one causes too many troubles I will implement it this way ^^

Reply:
I would rather suggest using that instead of making these jobs serial. It will just add more time and would not take advantage of the recently merged optimizations

Reply:
BTW. We are really stressing GA today. We have about 11 PRs + master builds now in the queue - and we had like 16 or 17. In total, we had ~ 70 builds today in 12 hours. This is quite a lot - usually we have that much in 24 hours so today seems to be a "peak" day. What adds up is that we disabled master canceling and we had 13 full master push builds today. Usually we have ca.  6-10 /day.

Reply:
Shouldn't that still be under our limit of 5000?

i.e 20 builds in an hour -> 20 *100 jobs per build = 2000

Reply:
> I would rather suggest using that instead of making these jobs serial. It will just add more time and would not take advantage of the recently merged optimizations

They will still use it, especially when there are more jobs in the queue. But let's see how canceling duplicate masters (already merged) will help it, and then we can see what's next. I would like to see this one working and see what effect it will have on a busy system though before i implement external canceling.

Reply:
BTW. I think what the real problem is now is that we have maybe 20 jobs running now not 150 - it looks like at least. so we might start being starved by other Apache projects like we had with Travis.

Reply:
I checked at Beaam: https://github.com/apache/beam and Skywalking: https://github.com/apache/skywalking where I helped to consult GA introduction and seems they also have a lot of delays. Opening issue to GA support.

Reply:
Ticket created: https://support.github.com/ticket/personal/0/867235